java-lib/src/main/java/com/wavefront/data/Validation.java

```diff
@@ -89,6 +89,22 @@ public static boolean charactersAreValid(String input) {
     return true;
   }
 
+  public static boolean logCharactersAreValid(String input) {
+    // Legal characters are 45-46 (-.), 48-57 (numbers), 65-90 (upper), 97-122 (lower), 95 (_)
```

Comment:
why are we doing this so low-level? I vote we do a high level check at either the char or string level.  Readability should trump perf unless we have a proven perf problem imo